From: =?UTF-8?q?Peter=20H=C3=BCwe?= <PeterHuewe@gmx.de>
Date: Tue, 21 May 2013 12:58:05 +0000
Subject: [PATCH v2 05/31] net/ethernet/broadcom/bnx2: Use module_pci_driver to
 register driver
Patch-mainline: v3.11-rc1

Git-commit: 5a4123f39ef14fc7c043f1611ed3b20b69325146
References: bsc#908686 FATE#317538


Removing some boilerplate by using module_pci_driver instead of calling
register and unregister in the otherwise empty init/exit functions.

Signed-off-by: Peter Huewe <peterhuewe@gmx.de>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Cho, Yu-Chen <acho@suse.com>
---
 drivers/net/ethernet/broadcom/bnx2.c |   16 +---------------
 1 file changed, 1 insertion(+), 15 deletions(-)

--- a/drivers/net/ethernet/broadcom/bnx2.c
+++ b/drivers/net/ethernet/broadcom/bnx2.c
@@ -8760,18 +8760,4 @@ static struct pci_driver bnx2_pci_driver
 	.err_handler	= &bnx2_err_handler,
 };
 
-static int __init bnx2_init(void)
-{
-	return pci_register_driver(&bnx2_pci_driver);
-}
-
-static void __exit bnx2_cleanup(void)
-{
-	pci_unregister_driver(&bnx2_pci_driver);
-}
-
-module_init(bnx2_init);
-module_exit(bnx2_cleanup);
-
-
-
+module_pci_driver(bnx2_pci_driver);
